### PR TITLE
Support issuing Latency commands

### LINKED ISSUE #1246: Support issuing Latency commands
Currently it is not possible to issue latency commands using jedis. You need to write your own client to issue custom commands.
There are a couple of latency commands (latency history, latency latest) that could be executed. ideally automatically guarded against the redis version (2.8.13 +)

### PR DESCRIPTION
**Pull request by Lcarrot**
Added LATENCY LATEST, LATENCY HISTORY and LATENCY RESET commands.

Resolves #1246

### CODE REVIEW

File: src/test/java/redis/clients/jedis/commands/jedis/ControlCommandsTest.java [hunk 1]
```diff
@@ -440,6 +442,24 @@ public void latencyDoctor() {
...
+  @Test
+  public void latencyHistoryFork() {
+    List<LatencyHistoryInfo> report = jedis.latencyHistory(Protocol.EventKeyword.FORK);
+    assertNotNull(report);
+  }
+
+  @Test
+  public void latencyReset() {
+    Long report = jedis.latencyReset();
+    assertNotNull(report);
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/commands/jedis/ControlCommandsTest.java:460]: It's actually `long` in method definition. So it won't be null anyway.
```suggestion
    assertTrue(jedis.latencyReset() >= 0);
```

File: src/main/java/redis/clients/jedis/CommandObjects.java [hunk 2]
```diff
@@ -4242,6 +4242,18 @@ public final CommandObject<Object> tFunctionCallAsync(String library, String fun
...
+  }
+
+  public final CommandObject<Long> latencyReset(Protocol.EventKeyword ... events) {
+    CommandArguments arguments = commandArguments(LATENCY).add(Keyword.RESET);
+    for (Protocol.EventKeyword event: events) {
+      arguments.add(event);
+    }
+    return new CommandObject<>(arguments, BuilderFactory.LONG);
+  }
+
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/CommandObjects.java:4256]: Remove these from here. Do the works in Jedis class.

File: src/main/java/redis/clients/jedis/commands/ServerCommands.java [hunk 3]
```diff
@@ -246,4 +250,10 @@ default void shutdown(SaveMode saveMode) throws JedisException {
    * @return the report
    */
   String latencyDoctor();
+
+  Map<String, LatencyLatestInfo> latencyLatest();
+
+  Object latencyHistory(Protocol.EventKeyword events);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/ServerCommands.java:256]:
```suggestion
  List<LatencyHistoryInfo> latencyHistory(Protocol.EventKeyword events);
```

File: src/main/java/redis/clients/jedis/commands/ServerCommands.java [hunk 4]
```diff
@@ -246,4 +250,10 @@ default void shutdown(SaveMode saveMode) throws JedisException {
...
+
+  long latencyReset(Protocol.EventKeyword ... events);
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/commands/ServerCommands.java:258]:
```suggestion
  long latencyReset(Protocol.EventKeyword... events);
```

File: src/main/java/redis/clients/jedis/Jedis.java [hunk 5]
```diff
@@ -9280,6 +9280,22 @@ public String latencyDoctor() {
...
+    connection.sendCommand(LATENCY, LATEST);
+    return BuilderFactory.LATENCY_LATEST_RESPONSE.build(connection.getOne());
+  }
+
+  public List<LatencyHistoryInfo> latencyHistory(EventKeyword event) {
+    checkIsInMultiOrPipeline();
+    return BuilderFactory.LATENCY_HISTORY_RESPONSE.build(connection.executeCommand(commandObjects.latencyHistory(event)));
+  }
+
+  public long latencyReset(EventKeyword ... events) {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/Jedis.java:9294]:
```suggestion
  public long latencyReset(EventKeyword... events) {
```

**Review by sazzad16 — CHANGES_REQUESTED** [review]

File: src/test/java/redis/clients/jedis/commands/jedis/ControlCommandsTest.java (hunk 1 above)
Comment (Lcarrot) [reply src/test/java/redis/clients/jedis/commands/jedis/ControlCommandsTest.java]: made a change like yours
make it by amend option

File: src/main/java/redis/clients/jedis/CommandObjects.java (hunk 2 above)
Comment (Lcarrot) [reply src/main/java/redis/clients/jedis/CommandObjects.java]: remove code from CommandObjects, now all works in Jedis class

File: src/main/java/redis/clients/jedis/commands/ServerCommands.java (hunk 3 above)
Comment (Lcarrot) [reply src/main/java/redis/clients/jedis/commands/ServerCommands.java]: fix type in interface

File: src/main/java/redis/clients/jedis/commands/ServerCommands.java (hunk 4 above)
Comment (Lcarrot) [reply src/main/java/redis/clients/jedis/commands/ServerCommands.java]: remove space

File: src/main/java/redis/clients/jedis/Jedis.java (hunk 5 above)
Comment (Lcarrot) [reply src/main/java/redis/clients/jedis/Jedis.java]: remove space

**Comment by Lcarrot** [comment]
i don't know can i resolve all treads if i fix it
So I left comments for all of them, please check if everything is ok now

**Comment by Lcarrot** [comment]
> i don't know can i resolve all treads if i fix it
> So I left comments for all of them, please check if everything is ok now

upd: i decided to resolve threads, i hope it's ok

File: src/main/java/redis/clients/jedis/Protocol.java [hunk 6]
```diff
@@ -371,4 +371,24 @@ public byte[] getRaw() {
       return raw;
     }
   }
+
+  public static enum EventKeyword implements Rawable {
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/Protocol.java:375]: Put this in `redis.clients.jedis.args` and the name should contain the word Latency; i.e. LatencyEvent.

**Review by sazzad16 — CHANGES_REQUESTED** [review]
Mostly LGTM!

File: src/main/java/redis/clients/jedis/Protocol.java (hunk 6 above)
Comment (Lcarrot) [reply src/main/java/redis/clients/jedis/Protocol.java]: add `redis.clients.jedis.args.LatencyEvent` instead of EventKeyword

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**